### PR TITLE
 Use bearer token as an authentication mechanism #73 

### LINKED ISSUE #73: Use bearer token as an authentication mechanism
ℹ️ This issue is part of Epic #55

## Objective
Use a Bearer token as an authorization mechanism for the API.

## Checklist
- [x] Remove the custom `auth-token` from the request header
- [x] Use a standard bearer token instead
- [x] Update the test suite

### PR DESCRIPTION
**Pull request by samu975**
## Description

This PR changed the get_token function in order to receive the bearer token.

-     Changed function get_token.
-     Uptdated account_controller_test.exs

Closes #73 

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by CristhianRodriguezMolina — CHANGES_REQUESTED** [review]
@samu975 Please ensure to get the feature branch from the branch you are gonna merge with, to ensure that the files and commits in the feature branch are up to date, for example, in this case, you have to get the feature branch like so:

```bash
> git checkout v0.3
> git checkout -b 73-feature-branch
```

File: lib/mintacoin_web/plugs/verify_api_token.ex [hunk 1]
```diff
@@ -44,8 +44,8 @@ defmodule MintacoinWeb.Plugs.VerifyApiToken do
 
   @spec get_token(conn :: conn()) :: token()
   defp get_token(conn) do
-    case get_req_header(conn, "auth-token") do
-      [token] -> token
+    case get_req_header(conn, "authorization") do
```
Comment (SanGeeky) [inline lib/mintacoin_web/plugs/verify_api_token.ex:47]: @samu975  please update this key with `Authorization` which is the standard for the authorization token.

**Review by SanGeeky — APPROVED** [review]

File: lib/mintacoin_web/plugs/verify_api_token.ex (hunk 1 above)
Comment (samu975) [reply lib/mintacoin_web/plugs/verify_api_token.ex:47]: Hi David, Authorization is the standard for the authorization token, but in this case, elixir only accepts header keys in lowercase, so if I change "authorization" to "Authorization," a Plug.Conn.InvalidHeaderError will appear

**Review by SanGeeky — CHANGES_REQUESTED** [review]
@samu975 after reviewing this PR I think we should use the word "authorization" instead of "authentication" for the other private functions and the error messages, to maintain code consistency. Do not forget to update the test also, thanks in advance.

**Review by SanGeeky — APPROVED** [review]

**Review by CristhianRodriguezMolina — APPROVED** [review]

**Review by SanGeeky — APPROVED** [review]

**Pull request merged**